Data/These.hs

```diff
@@ -194,6 +197,30 @@ catThat = mapMaybe justThat
 catThese :: [These a b] -> [(a, b)]
 catThese = mapMaybe justThese
 
+-- | Maybe map with Either like behavior on the left.
+mapMaybeThis :: Monoid c => (a -> Maybe b) -> These a c -> These b c
```

Comment:
The `Monoid` is used only for `mempty`. That's fishy.

Reply:
Its semantic with the way its done for `Either` and Ed Kmett approved! Learn all about it in my [library](http://hackage.haskell.org/package/compactable-0.1.2.2/docs/Control-Compactable.html) which already works with These to some degree, since Bifoldable and Bifunctor.